agent/src/main/java/io/pyroscope/javaagent/PyroscopeAgent.java

```diff
@@ -114,6 +114,9 @@ public Options build() {
                         scheduler = new SamplingProfilingScheduler(config, exporter, logger);
                     }
                 }
+                if (profiler == null) {
```

Comment:
should we add a setter for profiler field in the builder? Otherwise it looks like profiler is always null here